Harvesting into collection doesn't work (Xalan missing)

**Describe the bug**
DSpace Docker container version: dspace-7_x

The command `/dspace/bin/dspace harvest` doesn't work in the Docker.  
It throws exception: 
Exception in thread "main" java.lang.NoClassDefFoundError: org/apache/xpath/XPathAPI
        at ORG.oclc.oai.harvester2.verb.HarvesterVerb.getSingleString(HarvesterVerb.java:295)
        at ORG.oclc.oai.harvester2.verb.HarvesterVerb.getSingleString(HarvesterVerb.java:281)
        at ORG.oclc.oai.harvester2.verb.HarvesterVerb.harvest(HarvesterVerb.java:262)
        at ORG.oclc.oai.harvester2.verb.HarvesterVerb.<init>(HarvesterVerb.java:183)
        at ORG.oclc.oai.harvester2.verb.Identify.<init>(Identify.java:48)
        at org.dspace.harvest.HarvestedCollectionServiceImpl.verifyOAIharvester(HarvestedCollectionServiceImpl.java:200)
        at org.dspace.app.harvest.Harvest.pingResponder(Harvest.java:445)
        at org.dspace.app.harvest.Harvest.internalRun(Harvest.java:238)
        at org.dspace.scripts.DSpaceRunnable.run(DSpaceRunnable.java:104)
        at org.dspace.app.launcher.ScriptLauncher.executeScript(ScriptLauncher.java:149)
        at org.dspace.app.launcher.ScriptLauncher.handleScript(ScriptLauncher.java:131)
        at org.dspace.app.launcher.ScriptLauncher.main(ScriptLauncher.java:98)
Caused by: java.lang.ClassNotFoundException: org.apache.xpath.XPathAPI
        at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
        at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
        at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
        ... 12 more

_XPathAPI_ class should be in the _xalan-2.7.0.jar_ but this .jar is missing in the  _docker-cli_ image.

**To Reproduce**
Steps to reproduce the behavior:
1. docker pull dspace/dspace:dspace-7_x
2. docker run dspace/dspace:dspace-7_x
3. docker exec -it <CONTAINER_ID> /bin/bash
4. go to _/dspace/bin_
5. run command: `./dspace harvest -g -a http://lindat.mff.cuni.cz/repository/oai/request -i hdl_11234_3430`

**Expected behavior**
Start harvesting or throw error that Database or Solr is not working.

**Related work**

--- a/dspace-api/pom.xml
+++ b/dspace-api/pom.xml
@@ -550,9 +550,16 @@
             <groupId>com.ibm.icu</groupId>
             <artifactId>icu4j</artifactId>
         </dependency>
+        <!-- Codebase at https://github.com/OCLC-Research/oaiharvester2/ -->
         <dependency>
             <groupId>org.dspace</groupId>
             <artifactId>oclc-harvester2</artifactId>
+        </dependency>
+        <!-- Xalan is REQUIRED by 'oclc-harvester2' listed above (OAI harvesting fails without it).
+             Please do NOT use Xalan in DSpace codebase as it is not well maintained. -->
+        <dependency>
+            <groupId>xalan</groupId>
+            <artifactId>xalan</artifactId>
         </dependency>
         <dependency>
             <groupId>org.dspace</groupId>

--- a/pom.xml
+++ b/pom.xml
@@ -1567,10 +1567,18 @@
                 <artifactId>ojdbc6</artifactId>
                 <version>11.2.0.4.0</version>
             </dependency>
+            <!-- Codebase at https://github.com/OCLC-Research/oaiharvester2/ -->
             <dependency>
                 <groupId>org.dspace</groupId>
                 <artifactId>oclc-harvester2</artifactId>
                 <version>0.1.12</version>
+            </dependency>
+            <!-- Xalan is REQUIRED by 'oclc-harvester2' listed above (OAI harvesting fails without it).
+                 Please do NOT use Xalan in DSpace codebase as it is not well maintained. -->
+            <dependency>
+                <groupId>xalan</groupId>
+                <artifactId>xalan</artifactId>
+                <version>2.7.2</version>
             </dependency>
             <dependency>
                 <groupId>org.apache.httpcomponents</groupId>

